Documentation/ApiOverview/Localization/LocalizationApi/LanguageService.rst

```diff
@@ -12,5 +12,37 @@ be created directly, therefore its constructor is internal. Create a
 
 In the backend context a :php:`LanguageService` is stored in the global
 variable :php:`$GLOBALS['LANG']`.
+In the frontend a :php:`LanguageService` can be accessed via the contentObject:
+
+..  code-block:: php
+    :caption: Classes/Controller/ExampleController.php
+
+    use Psr\Http\Message\ServerRequestInterface;
+    use TYPO3\CMS\Core\Localization\LanguageServiceFactory;
+    use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+    class ExampleController {
+        proteced ServerRequestInterface $request;
+
+        public function processAction(ServerRequestInterface $request): string
+            $this->request = $request;
+
+            $content = '';
+            ...
+            $label = $this->getTranslatedLabel('LLL:EXT:my_extension/Resources/Private/Language/locallang.xlf:labels.exampleLabel');
+            ...
+
+            return $content;
+        }
+
+        protected function getTranslatedLabel(string $key): string
+        {
+            $language = $this->request->getAttribute('language') ?? $this->request->getAttribute('site')->getDefaultLanguage();
+            $languageService = GeneralUtility::makeInstance(LanguageServiceFactory::class)
```

Comment:
As this is not wrong, LanguageServiceFactory should be injected into constructor IMHO as best practise.